sdwmbar.c

```diff
@@ -89,6 +90,21 @@ getbatt(int *buff)
 	}
 }
 
+void
+getac(char *ac)
+{
+	int buff;
+	size_t len = sizeof(buff);
+	sysctlbyname("hw.acpi.acline", &buff, &len, NULL, 0);
+
+	if (buff == 1) {
+		*ac = '+';
+		*(ac+1) = '\0';
+	} else {
+		*ac = '\0';
+	}
+}
```

Comment:
Could be streamlined to

`*buff = (ac) ? "+" : ""`

Reply:
Add null termination afterwards, of course. 
Or assign the array in `main()` as `char ac[AC_LEN] = {0}`.